baldrick/plugins/github_towncrier_changelog.py

```diff
@@ -146,31 +149,34 @@ def process_towncrier_changelog(pr_handler, repo_handler, headers):
 
     modified_files = pr_handler.get_modified_filenames()
 
-    messages = []
+    messages = {}
+
     matching_file = check_sections(modified_files, section_dirs)
-    if not matching_file:
-        messages.append(cl_config.get("missing_file_message", NO_CHANGELOG))
+
+    if skip_label and skip_label in pr_handler.labels:
+
+        messages['missing_file'] = 'Skipping changelog presence test', True
```

Comment:
Surely we don't want statuses for skipped checks?!

Reply:
I'm not sure how to deal with this case, since maybe those statuses have been set in a previous run - so how do we remove them? Should a status of 'None' cause the status to be actively removed?

Reply:
:thinking: So this is only a problem for statues on commits where we have already posted a fail status right? Can we implement a way of getting the existing contexts on a sha? So we only post this status if we need to clear a old one. That way we don't post pointless statues on new commits?

Reply:
Sounds good to me, will try and work on this shortly

Reply:
Fixed - basically if a status is not returned but was returned before, we just set the status to ``success`` and set the message to ``This check has been skipped`` automatically.